TST : unhandle exceptions raised because of Non closed temporary file streams

## Environment

side effect of #2379 

```bash
$ python -m platform
Windows-10-10.0.22621-SP0
python 3.10

$ python -c "import pypdf;print(pypdf._debug_versions)"
pypdf==3.17.4, crypt_provider=('cryptography', '41.0.1'), PIL=10.1.0
```

## Code + PDF

```cmd
pytest -k test_writer_operations_by_traditional_usage
```

## Traceback

This is the complete traceback I see:

```
E               pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO [closed]>
E
E               Traceback (most recent call last):
E                 File "E:\PyPDF4.git\PyPDF2\tests\test_writer.py", line 227, in test_writer_operations_by_traditional_usage
E                   write_data_here = convert(NamedTemporaryFile(suffix=".pdf", delete=needs_cleanup).name)
E               ResourceWarning: unclosed file <_io.BufferedRandom name=15>

C:\Python310\lib\site-packages\_pytest\unraisableexception.py:78: PytestUnraisableExceptionWarning```


## Analysis

the temporary file created is not closed : it should be

--- a/tests/test_writer.py
+++ b/tests/test_writer.py
@@ -224,9 +224,8 @@ def writer_operate(writer: PdfWriter) -> None:
 )
 def test_writer_operations_by_traditional_usage(convert, needs_cleanup):
     if callable(convert):
-        fo = NamedTemporaryFile(suffix=".pdf", delete=False)
-        write_data_here = convert(fo.name)
-        fo.close()
+        with NamedTemporaryFile(suffix=".pdf", delete=False) as fo:
+            write_data_here = convert(fo.name)
     else:
         write_data_here = convert
 
@@ -256,9 +255,8 @@ def test_writer_operations_by_traditional_usage(convert, needs_cleanup):
 )
 def test_writer_operations_by_semi_traditional_usage(convert, needs_cleanup):
     if callable(convert):
-        fo = NamedTemporaryFile(suffix=".pdf", delete=False)
-        write_data_here = convert(fo.name)
-        fo.close()
+        with NamedTemporaryFile(suffix=".pdf", delete=False) as fo:
+            write_data_here = convert(fo.name)
     else:
         write_data_here = convert
 
@@ -287,9 +285,8 @@ def test_writer_operations_by_semi_traditional_usage(convert, needs_cleanup):
 )
 def test_writer_operations_by_semi_new_traditional_usage(convert, needs_cleanup):
     if callable(convert):
-        fo = NamedTemporaryFile(suffix=".pdf", delete=False)
-        write_data_here = convert(fo.name)
-        fo.close()
+        with NamedTemporaryFile(suffix=".pdf", delete=False) as fo:
+            write_data_here = convert(fo.name)
     else:
         write_data_here = convert
 
@@ -313,9 +310,8 @@ def test_writer_operations_by_semi_new_traditional_usage(convert, needs_cleanup)
 )
 def test_writer_operation_by_new_usage(convert, needs_cleanup):
     if callable(convert):
-        fo = NamedTemporaryFile(suffix=".pdf", delete=False)
-        write_data_here = convert(fo.name)
-        fo.close()
+        with NamedTemporaryFile(suffix=".pdf", delete=False) as fo:
+            write_data_here = convert(fo.name)
     else:
         write_data_here = convert
 

--- a/tests/test_writer.py
+++ b/tests/test_writer.py
@@ -224,7 +224,9 @@ def writer_operate(writer: PdfWriter) -> None:
 )
 def test_writer_operations_by_traditional_usage(convert, needs_cleanup):
     if callable(convert):
-        write_data_here = convert(NamedTemporaryFile(suffix=".pdf", delete=False).name)
+        fo = NamedTemporaryFile(suffix=".pdf", delete=False)
+        write_data_here = convert(fo.name)
+        fo.close()
     else:
         write_data_here = convert
 
@@ -254,7 +256,9 @@ def test_writer_operations_by_traditional_usage(convert, needs_cleanup):
 )
 def test_writer_operations_by_semi_traditional_usage(convert, needs_cleanup):
     if callable(convert):
-        write_data_here = convert(NamedTemporaryFile(suffix=".pdf", delete=False).name)
+        fo = NamedTemporaryFile(suffix=".pdf", delete=False)
+        write_data_here = convert(fo.name)
+        fo.close()
     else:
         write_data_here = convert
 
@@ -281,11 +285,11 @@ def test_writer_operations_by_semi_traditional_usage(convert, needs_cleanup):
         (BytesIO(), False),
     ],
 )
-def test_writer_operations_by_semi_new_traditional_usage(
-    convert, needs_cleanup
-):
+def test_writer_operations_by_semi_new_traditional_usage(convert, needs_cleanup):
     if callable(convert):
-        write_data_here = convert(NamedTemporaryFile(suffix=".pdf", delete=False).name)
+        fo = NamedTemporaryFile(suffix=".pdf", delete=False)
+        write_data_here = convert(fo.name)
+        fo.close()
     else:
         write_data_here = convert
 
@@ -309,7 +313,9 @@ def test_writer_operations_by_semi_new_traditional_usage(
 )
 def test_writer_operation_by_new_usage(convert, needs_cleanup):
     if callable(convert):
-        write_data_here = convert(NamedTemporaryFile(suffix=".pdf", delete=False).name)
+        fo = NamedTemporaryFile(suffix=".pdf", delete=False)
+        write_data_here = convert(fo.name)
+        fo.close()
     else:
         write_data_here = convert
 

